src/main/java/org/opensearch/knn/index/query/DefaultKNNWeight.java

```diff
@@ -192,6 +193,9 @@ protected NativeMemoryAllocation loadGraph(
         final String modelId,
         LeafReaderContext context
     ) throws ExecutionException, IOException {
+        KnnVectorValues knnVectorValues = (vectorDataType == VectorDataType.BYTE)
+            ? context.reader().getByteVectorValues(knnQuery.getField())
+            : context.reader().getFloatVectorValues(knnQuery.getField());
```

Comment:
All the places where you are using Lucene KNNVectorValues please use our own vector values: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/vectorvalues/KNNVectorValues.java and this factory class to create the vector values: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/vectorvalues/KNNVectorValuesFactory.java

Please remove all the other references of Lucene KNNVectorValues